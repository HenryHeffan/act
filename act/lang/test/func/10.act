function g (pint x) : pint
{
  pint w[10];
  chp {
     w[5] := x*x;
     self := w[5]
  }
}

bool x[1+g(0)];

bool y[1];

x = y;

bool z[1+g(2)];

y = z;
