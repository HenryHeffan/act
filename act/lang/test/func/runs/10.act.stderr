In expanding ::<Global>
Connection: y = z
  LHS: bool[1]
  RHS: bool[5]
FATAL: Type-checking failed on connection
	Types `bool[1]' and `bool[5]' are not compatible
